CDP4Composition/Mvvm/BrowserViewModelBase.cs

```diff
@@ -1044,5 +1058,47 @@ private void PopulateCreateContextMenu()
             this.CreateContextMenu.AddRange(this.ContextMenu.Where(x => x.MenuItemKind == MenuItemKind.Create));
             this.IsAddButtonEnabled = this.CreateContextMenu.Any();
         }
+
+        /// <summary>
+        /// Executes a long running <see cref="Action"/> using the current <see cref="Dispatcher"/> async.
+        /// </summary>
+        /// <param name="action">The <see cref="Action"/></param>
+        /// <param name="loadingMessage">A message to be shown during execution.</param>
+        protected void ExecuteLongRunningDispatcherAction(Action action, string loadingMessage = "")
```

Comment:
should this not return a Task instead of void?

Reply:
No, because nothing should be awaited. 
The trick is that we do a "fire and forget" invocation of some code and we prioritize the code to be executed in the Dispatcher stack (DispatcherPriority.Background). 
Because of the lower priority, the Dispatcher handles rendering the loading control (IsBusy = true) first. 
When the delayed code starts executing (on the UI thread) the code will block the UI thread again.

If you use the Dispatcher extensively, you're in deep s***, but for actions like this it works perfectly.

Reply:
ok